torch/fft/__init__.py

```diff
@@ -9,18 +9,283 @@
+the output contains only the positive frequencies below the Nyquist frequency.
+To compute the full output, use :func:`~torch.fft.fft`
+
+Args:
+    input (Tensor): the real input tensor
+    n (int, optional): Signal length. If given, the input will either be zero-padded
+        or trimmed to this length before taking the real FFT.
+    dim (int, optional): The dimension along which to take the one dimensional real FFT.
+    norm (str, optional): Normalization mode. Between the forward and backward
+        transforms (:func:`~torch.fft.rfft` and :func:`~torch.fft.irfft`) a total
+        normalization factor of ``1/n`` is applied. The mode determines where
+        this normalization is applied:
+
+        * ``"forward"`` - the forward transform is normalized by ``1/n``, backward is unnormalized.
+        * ``"backward"`` - the backward transform is normalized by ``1/n``, forward is unnormalized.
+        * ``"ortho"`` - both transforms are normalized by ``1/sqrt(n)`` (making it orthonormal).
+
+        Default is ``"backward"``.
+
+Example:
+
+    >>> import torch.fft
+    >>> t = torch.arange(4)
+    >>> t
+    tensor([0, 1, 2, 3])
+    >>> torch.fft.rfft(t)
+    tensor([ 6.+0.j, -2.+2.j, -2.+0.j])
+
+    Compare against the full output from :func:`~torch.fft.fft`:
+
+    >>> torch.fft.fft(t)
+    tensor([ 6.+0.j, -2.+2.j, -2.+0.j, -2.-2.j])
+
+    Notice that the symmetric element ``T[-1] == T[1].conj()`` is omitted.
+    At the Nyquist frequency ``T[-2] == T[2].conj() == T[2]`` is given but must
+    always be purely real.
+""")
+
+irfft = _add_docstr(_fft.fft_irfft, r"""
+irfft(input, n=None, dim=-1, norm=None) -> Tensor
+
+Computes the inverse of :func:`~torch.fft.rfft`.
+
+:attr:`input` is a one-sided Hermitian signal in the Fourier domain, as
+produced by :func:`~torch.fft.rfft`. By the Hermitian property, the time-domain
+signal will be purely real.
+
+Note:
+    Some input frequencies must be purely real to satisfy the Hermitian
+    property. In these cases the imaginary component will be ignored.
+
+Note:
+    The correct interpretation of the hermitian input depends on the length of
+    the original data, as given by :attr:`n`. This is because each input shape
+    could correspond to either an odd or even length signal. By default, the
+    signal is assumed to be even length and odd signals will not round-trip
+    properly. So, it is recommended to always pass the signal length :attr:`n`:
+
+Args:
+    input (Tensor): the input tensor representing a half-Hermitian signal
+    n (int, optional): Output signal length. This determines the length of the
+        output signal. If given, the input will either be zero-padded or trimmed to this
+        length before taking the real IFFT.
+        Defaults to even output: ``n=2*(input.size(dim) - 1)``.
+    dim (int, optional): The dimension along which to take the one dimensional real IFFT.
+    norm (str, optional): Normalization mode. Between the forward and backward
```

Comment:
The more I read this normalization mode description the more I appreciate it, but I still want to try to tweak it slightly per the above.